bin/hbase-daemon.sh

```diff
@@ -78,22 +78,33 @@ hbase_rotate_log ()
     fi
 }
 
-cleanAfterRun() {
-  if [ -f ${HBASE_PID} ]; then
-    # If the process is still running time to tear it down.
-    kill -9 `cat ${HBASE_PID}` > /dev/null 2>&1
-    rm -f ${HBASE_PID} > /dev/null 2>&1
+function sighup_handler
+{
+  # pass through SIGHUP if we can
+  if [ -f "${HBASE_PID}" ] ; then
+    kill -s HUP "$(cat "${HBASE_PID}")"
```

Comment:
You'd rather see the output than /dev/null it? And does the old pid stick around now?

Reply:
Testing in a docker container that disappears... I'm guessing that this script already handles the pid file, but let me look more closely...

Reply:
Oh. `cleanupAfterRun` handles the pid file.